slsDetectorSoftware/src/DetectorImpl.cpp

```diff
@@ -1281,6 +1303,58 @@ void DetectorImpl::startAcquisition(bool blocking, std::vector<int> positions) {
     }
 }
 
+void DetectorImpl::sendSoftwareTrigger(const bool block,
+                                       std::vector<int> positions) {
+    bool handleSynchronization = false;
+    // multi module sync enabled jungfrau
+    if (size() > 1 && (shm()->detType == defs::JUNGFRAU) &&
+        (Parallel(&Module::getSynchronizationFromStopServer, positions)
+             .tsquash("Inconsistent synchronization among modules"))) {
+        handleSynchronization = true;
+    }
+
+    // slaves first
+    if (handleSynchronization) {
+        std::vector<int> master;
+        std::vector<int> slaves;
+        getMasterSlaveList(positions, master, slaves);
+        if (!slaves.empty())
+            Parallel(&Module::sendSoftwareTrigger, slaves, false);
+        if (!master.empty())
+            Parallel(&Module::sendSoftwareTrigger, master, block);
+    }
+    // all in parallel
+    else {
+        Parallel(&Module::sendSoftwareTrigger, positions, block);
+    }
+}
+
+void DetectorImpl::stopDetector(std::vector<int> positions) {
+
+    bool handleSynchronization = false;
```

Comment:
Could we make a function for the handle synchronization logic? I see it in two places here. Risk is that we only change one if we upgrade =)

Reply:
I wanted to do that as well, but they are quite different for each of the functions. For the software trigger, its not for M3. For the stop, its slaves first. Or is there another way to handle this? I couldnt find a way without cluttering.